### PR TITLE
feat: implementing Storage Billing model property to BQ Datasets

### PR DESCRIPTION
**Pull request by sriramsridhar**
Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:
- [x] Make sure to open an issue as a [bug/issue](https://github.com/googleapis/python-bigquery/issues/new/choose) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea
- [x] Ensure the tests and linter pass
- [x] Code coverage does not decrease (if any source code was changed)
- [x] Appropriate docs were updated (if necessary)

Fixes Issue #1639 🦕

### CODE REVIEW

**Comment by tswast** [comment]
Thanks so much for the contribution @sriramsridhar !

**Review by tswast — APPROVED** [review]

**Comment by sriramsridhar** [comment]
@obada-ab Please review this pr and share your comments

**Comment by sriramsridhar** [comment]
@tswast  Please merge this PR

File: google/cloud/bigquery/dataset.py [hunk 1]
```diff
@@ -763,6 +764,25 @@ def default_encryption_configuration(self, value):
...
+
+        Raises:
+            ValueError: for invalid value types.
+        """
+        return self._properties.get("storageBillingModel")
+
+    @storage_billing_model.setter
+    def storage_billing_model(self, value):
+        if not isinstance(value, str) and value is not None:
+            raise ValueError("Pass a string, or None")
```
Comment (tswast) [inline google/cloud/bigquery/dataset.py:780]: We're missing some unit tests for some of these branches:

```
google/cloud/bigquery/dataset.py                               352      2     81      3    99%   780, 781->783, 784
```

**Review by tswast — CHANGES_REQUESTED** [review]

File: google/cloud/bigquery/dataset.py (hunk 1 above)
Comment (agonencyto) [reply google/cloud/bigquery/dataset.py:780]: any ETA on this ?

**Comment by SudoSpartanDan** [comment]
Is there an ETA on this feature? Our team is waiting on this to be able to set this attribute.

**Review by tswast — APPROVED** [review]

**Pull request merged**